File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        count = 0
        lengths = []
        for index, num in enumerate(nums):
            if num == 1:
                count += 1
            else:
                lengths.append(count)
                count = 0
            lengths.append(count)
        return max(lengths)

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-0.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int sum = nums[0], max = 0;
        for (int i = 1; i < nums.length; i++){
            if (nums[i-1] < nums[i]){
                sum += nums[i];
            } else {
                max = Math.max(max, sum);
                sum = nums[i];
            }
        }
        max = Math.max(max, sum);
        return max;
    }
}
